cannon/testdata/example/mt-map/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"sync"
+)
+
+func main() {
```

Comment:
Yes, Some tests not running successfully. We could further deep dive them.

Reply:
Yeah, it would be helpful to go ahead and port them over.  If you can't get them running, you can skip them for now and we can take a closer look.  If you do skip, please leave a message referencing this issue:
```
t.Skip("TODO(#12162) Investigate test failure")
```

Reply:
Got it @mbaxter, will do that after devcon

Reply:
Just a heads up - we are working against an internal deadline so I started looking into porting more of the Go `sync` tests over today: https://github.com/optimism-java/optimism/compare/moresynctests...mbaxter:review/optimism-java/moresynctests

I will open a PR to your branch when I finish porting these over.

Reply:
thanks.